### PR TITLE
Implement of "Order by" for "Folder view type"

### PR DESCRIPTION
**Pull request by mirtoto**
- Supporting order files by name (asc/desc) & time (asc/desc).
- Folders are always put on beginning of the list/grid.
- Fixed advanced options save settings for "Folder view type".
- Simplified code connected with generating table for "Folder view type".
- Added missed phrases in order to be translatable.
- Renamed CSS classes connected with "Folder view type" in order to be more unique.
- Improved CSS of "Google Drive Documents" block to be more compliant with WordPress admin theme.
- Minor fixes of default CSS for "Folder view type".

### CODE REVIEW

**Comment by marekdedic** [comment]
Hi,
thanks for the PR. Since it isn't separated from #67, I'm going to wait until that is merged to review this one.

**Comment by marekdedic** [comment]
Ok, could you please rebase/merge the current master to keep it clean, then let's review it. :)

**Comment by marekdedic** [comment]
Somehow the PR still contains changes which are already in master (for example in `.gitignore`) - Could you please do `git merge upstream/master`?

**Comment by marekdedic** [comment]
Perfect, now there's only the actual changes, thanks!

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [inline plugin/admin/options/option-types/class-selectfield.php:65]: This is an ugly hack. Granted, it was an ugly hack even before. Could you please change it so that the available options are passed as a parameter in the constructor? Or subclass it...

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [inline plugin/admin/options/option-types/class-selectfield.php:40]: Yeah, similarly to the comment below, this should validate the options. If they're passed in the constructor, that shouldn't be too hard.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [inline plugin/admin/options/option-types/class-selectfield.php:46]: This is all incorrect, am I getting it right? It's verifying against values which are not even in the options (there should be `list` and `grid`)

File: plugin/public/js/select-setting.js
Comment (marekdedic) [inline plugin/public/js/select-setting.js:29]: Again the same hack. Could we pass the options from PHP dynamically?

File: plugin/public/block.php
Comment (marekdedic) [inline plugin/public/block.php:128]: I am getting
```php
Notice:  Undefined variable: order_by in /home/user/www/wp-content/plugins/skaut-google-drive-documents/public/block.php on line 130
```

```

File: plugin/public/css/block.css
Comment (marekdedic) [inline plugin/public/css/block.css:2]: Is this variable provided by WordPress? I couldn't find any documentation for it...

If so, would it be available on older WordPress versions?

File: plugin/public/css/block.css
Comment (marekdedic) [inline plugin/public/css/block.css]: Eh, I'm not a fan. Just color looks better I think.

**Review by marekdedic — CHANGES_REQUESTED** [review]
Hi, thanks again for the PR! I have some minor points, but there's also one big issue - the different selects are hard-coded all over the place. That means to add any select in the future, you'd need to crawl through everything that uses it and that's just bad code design. I thinks this should be fixable either by passing the available options to the constructor, or by subclassing. Could you please take a look at it? Thanks!

File: plugin/admin/options/option-types/class-selectfield.php
Comment (mirtoto) [reply plugin/admin/options/option-types/class-selectfield.php:40]: There is no big sense to validate values returned by checkboxes. It will be always valid because users can't modify its values.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (mirtoto) [reply plugin/admin/options/option-types/class-selectfield.php:46]: Yes, you right - this is wrong code.

File: plugin/public/css/block.css
Comment (mirtoto) [reply plugin/public/css/block.css:2]: This CSS variable is connected with WordPress Admin Color Scheme and it was included to core with WordPress 3.8.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [reply plugin/admin/options/option-types/class-selectfield.php:46]: Ok, well, thanks for noticing & fixing it.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [reply plugin/admin/options/option-types/class-selectfield.php:40]: Hmm, I was thinking about the users not being able to deliberately break the plugin or insert XSS or something... But true, this would protect against XSS and anyway, this should only be editable by administrators....

File: plugin/public/css/block.css
Comment (marekdedic) [reply plugin/public/css/block.css:2]: Oh, but it just occured to me that there is no preprocessing done on the CSS so this would break on IE :/ COuld we please use something with better compatibility?

Comment (mirtoto) [reply plugin/public/css/block.css:2]: I can add fallback color but IE is not fully supported by WordPress (when you enter to WordPress admin panel it advice you change browser to more secure). Apart this, when I try to edit page with this plugin in IE (I was surprised that I still have this browser installed) I see message that "this webpage doesn't support block "Files and Docs from Google Drive" - so anyway it is not working on admin side in IE.

File: plugin/public/block.php
Comment (mirtoto) [reply plugin/public/block.php:128]: Ok. I will fix it.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (mirtoto) [reply plugin/admin/options/option-types/class-selectfield.php:65]: Yes - it is ugly but much better anyway than previous code :) I can improve this but not in this PR - I already done some modification in base class of these settings and it will be easier to modify it after applying these modifications. We can mark this somehow as TODO work.

File: plugin/public/css/block.css
Comment (mirtoto) [reply plugin/public/css/block.css]: As you want - it may be also added in user CSS so it is no problem for me. Or maybe can we find some nice icon with folder?

File: plugin/public/css/block.css
Comment (marekdedic) [reply plugin/public/css/block.css:2]: Well, then let's not bother if it doesn't work anyway :(

File: plugin/public/css/block.css
Comment (marekdedic) [reply plugin/public/css/block.css]: Well, an icon would be the best solution I think...

Comment (mirtoto) [reply plugin/public/css/block.css]: I was doing some research and there is possibility to use WordPress icon: [https://developer.wordpress.org/resource/dashicons/#open-folder](url) so I modify code to use these dashicons.

Comment (marekdedic) [reply plugin/public/css/block.css]: That's perfect, thanks.

File: plugin/admin/options/option-types/class-selectfield.php
Comment (marekdedic) [reply plugin/admin/options/option-types/class-selectfield.php:65]: Hmm, ok. If this were my codebase with my standards, I would say no (because a promise to fix bad code later is almost never fulfilled :D ). But given the really bad code quality here, I've filed it as #70 and let's solve it later.

File: plugin/public/js/select-setting.js
Comment (marekdedic) [reply plugin/public/js/select-setting.js:29]: See #70

**Review by marekdedic — APPROVED** [review]

**Comment by marekdedic** [comment]
The folder icons could use some more vertical aligning (they seem too low) and more space between them and the text, but I'm merging this as is, that can be done later.

**Pull request merged**
